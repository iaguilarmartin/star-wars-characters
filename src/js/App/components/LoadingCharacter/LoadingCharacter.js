import React from 'react';
import styled from '@emotion/styled';

import fonts from '../../styles/fonts';

const Section = styled.section`
  flex-grow: 1;
  position: relative;
  text-align: center;
`;

const LoadingMeassage = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%);
  font-family: ${fonts.Starjedi};
`;

const LoadingCharacter = () => (
  <Section>
    <LoadingMeassage>Loading character information...</LoadingMeassage>
  </Section>
);

export default LoadingCharacter;
