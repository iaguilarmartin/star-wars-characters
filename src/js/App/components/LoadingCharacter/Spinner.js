import React from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';

import colors from '../../styles/colors';

const circleFadeDelay = keyframes`
  0%, 39%, 100% { opacity: 0; }
  40% { opacity: 1; }
`;

const Container = styled.div`
  position: relative;
  height: 48px;
  width: 48px;

  div {
    height: 100%;
    width: 100%;
    position: absolute;
    left: 0;
    top: 0;

    &:before {
      content: '';
      display: block;
      margin: 0 auto;
      height: 15%;
      width: 15%;
      background-color: ${colors.yellow};
      border-radius: 100%;
      animation: ${circleFadeDelay} 1.2s infinite ease-in-out both;
    }
  }

  .circle2 {
    transform: rotate(30deg);
    &:before {
      animation-delay: -1.1s;
    }
  }
  .circle3 {
    transform: rotate(60deg);
    &:before {
      animation-delay: -1s;
    }
  }
  .circle4 {
    transform: rotate(90deg);
    &:before {
      animation-delay: -0.9s;
    }
  }
  .circle5 {
    transform: rotate(120deg);
    &:before {
      animation-delay: -0.8s;
    }
  }
  .circle6 {
    transform: rotate(150deg);
    &:before {
      animation-delay: -0.7s;
    }
  }
  .circle7 {
    transform: rotate(180deg);
    &:before {
      animation-delay: -0.6s;
    }
  }
  .circle8 {
    transform: rotate(210deg);
    &:before {
      animation-delay: -0.5s;
    }
  }
  .circle9 {
    transform: rotate(240deg);
    &:before {
      animation-delay: -0.4s;
    }
  }
  .circle10 {
    transform: rotate(270deg);
    &:before {
      animation-delay: -0.3s;
    }
  }
  .circle11 {
    transform: rotate(300deg);
    &:before {
      animation-delay: -0.2s;
    }
  }
  .circle12 {
    transform: rotate(330deg);
    &:before {
      animation-delay: -0.1s;
    }
  }
`;

const StyledSpinner = () => (
  <Container>
    <div className="circle1" />
    <div className="circle2" />
    <div className="circle3" />
    <div className="circle4" />
    <div className="circle5" />
    <div className="circle6" />
    <div className="circle7" />
    <div className="circle8" />
    <div className="circle9" />
    <div className="circle10" />
    <div className="circle11" />
    <div className="circle12" />
  </Container>
);

export default StyledSpinner;
