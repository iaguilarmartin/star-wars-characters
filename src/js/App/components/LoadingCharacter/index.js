import LoadingCharacter from './LoadingCharacter';

export default LoadingCharacter;
