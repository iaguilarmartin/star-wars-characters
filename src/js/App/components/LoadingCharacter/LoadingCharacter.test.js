import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LoadingCharacter from './LoadingCharacter';

describe('<LoadingCharacter/>', () => {
  it('renders', () => {
    shallow(<LoadingCharacter />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<LoadingCharacter />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
