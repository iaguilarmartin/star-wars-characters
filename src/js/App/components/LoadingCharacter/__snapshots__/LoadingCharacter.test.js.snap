// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingCharacter/> matches the snapshot 1`] = `
<Section>
  <LoadingAnimation
    className={null}
  />
  Loading character information...
</Section>
`;
